wallet/walletrpc.c

```diff
@@ -474,7 +474,6 @@ static void json_listaddrs(struct command *cmd,
 	}
 	json_array_end(response);
 	json_object_end(response);
-	tal_free(tmpctx);
```

Comment:
You also need to remove the `tal_free(tmpctx)` calls in the failure routes.  Note that `json_get_params` itself will call `command_fail_detailed` if it returns false, and since `tmpctx` is now rooted from `cmd`, you should not be calling `tal_free(tmpctx)` there either.  Also holds for your uses of `command_fail`.

Reply:
Thanks, I understand and fix.